mods/default/functions.lua

```diff
@@ -111,6 +111,21 @@ minetest.register_abm({
 
 
 --
+-- optimized helper to put all items in an inventory into a drops list
+--
+function default.get_inventory_drops(pos, inventory, drops)
+	local inv = minetest.get_meta(pos):get_inventory()
+	local n = #drops
+	for i = 1, inv:get_size(inventory) do
+		local stack = inv:get_stack(inventory, i)
+		if stack:get_count() > 0 then
+			drops[n+1] = stack:to_string()
```

Comment:
No, that would break the ability for this function to append drops - it was written so that it can be called multiple times - each time adding more items but not overwriting existing entries already in drops[]